proto/properties.go

```diff
@@ -596,13 +603,52 @@ func MessageType(name string) reflect.Type {
 
 // A registry of all linked proto files.
 var (
-	protoFiles = make(map[string][]byte) // file name => fileDescriptor
+	protoFiles        = make(map[string][]byte) // file name => fileDescriptor
+	gogoProtoRegistry = new(protoregistry.Files)
```

Comment:
Maybe this can actually be made public like `*GlobalFiles` and then other checks can be performed by the caller if needed?

Reply:
Exported this as `GogoResolver`. Let me know if you see any issue with that

Reply:
I think it should be okay exported.